engine/src/main/java/org/terasology/logic/players/LocalPlayerSystem.java

```diff
@@ -212,6 +215,33 @@ private void standPlayer(EntityRef entity) {
         clientComp.character.send(new SetMovementModeEvent(MovementMode.WALKING));
     }
 
+    /**
+     * Auto move is disabled every time any other movement event occurs.
+     */
+    private void stopAutoMove() {
+        isAutoMove = false;
+    }
+
+    /**
+     * Start a thread that moves the entity in the camera direction at every period of 0.2 seconds.
+     * @param entity
+     */
+    private void startAutoMove(EntityRef entity) {
+        isAutoMove = true;
+        ClientComponent clientComponent = entity.getComponent(ClientComponent.class);
+        new Thread(() -> {
```

Comment:
Imho this should not be done with a new Thread, but with the help of `DelayManager`. You can schedule events there, and cancel the periodic events if auto-move should stop.